pkg/manifest/anaconda_installer_iso_tree.go

```diff
@@ -560,3 +556,28 @@ func makeISORootPath(p string) string {
 	fullpath := path.Join("/run/install/repo", p)
 	return fmt.Sprintf("file://%s", fullpath)
 }
+
+func makeKickstartSudoersPost(names []string) string {
```

Comment:
Having a test for this would be nice, maybe something like?
```go
func TestMakeKickstartSudoersPostEmpty(t *testing.T) {
	assert.Equal(t, "", makeKickstartSudoersPost(nil))
}

func TestMakeKickstartSudoersPost(t *testing.T) {
	assert.Equal(t, "...", makeKickstartSudoersPost([]string{"foo", "foo", "bar"}))
}
```
which makes me wonder if makeKickstartSudoersPost() for the empty list should just return "" and wit hthat I wonder if we should do something like:
```go
diff --git a/pkg/manifest/anaconda_installer_iso_tree.go b/pkg/manifest/anaconda_installer_iso_tree.go
index ba0ae9db3..ae59a9936 100644
--- a/pkg/manifest/anaconda_installer_iso_tree.go
+++ b/pkg/manifest/anaconda_installer_iso_tree.go
@@ -534,10 +534,10 @@ func (p *AnacondaInstallerISOTree) makeKickstartStages(kickstartOptions *osbuild
 
        stages = append(stages, osbuild.NewKickstartStage(kickstartOptions))
 
-       if len(p.NoPasswd) > 0 {
+       hardcodedKickstartBits := makeKickstartSudoersPost(p.NoPasswd)
+       if hardcodedKickstartBits != "" {
                // Because osbuild core only supports a subset of options,
                // we append to the base here with hardcoded wheel group with NOPASSWD option
-               hardcodedKickstartBits := makeKickstartSudoersPost(p.NoPasswd)
                kickstartFile, err := kickstartOptions.IncludeRaw(hardcodedKickstartBits)
                if err != nil {
                        panic(err)

```
but that is mostly idle wondering - I still think a test for the helper is easy and nice :)

Reply:
I'm adding a line to the existing test that checks for the expected inline ID, which is a hash of the content.

Reply:
Maybe I'm missing something, I would love to see a test for the generated kickstart file content but maybe that is covered already and I don't see it (sorry!). As an experiment I did:
```go
diff --git a/pkg/manifest/anaconda_installer_iso_tree.go b/pkg/manifest/anaconda_installer_iso_tree.go
index ba0ae9db3..9189dc72e 100644
--- a/pkg/manifest/anaconda_installer_iso_tree.go
+++ b/pkg/manifest/anaconda_installer_iso_tree.go
@@ -558,6 +558,8 @@ func makeISORootPath(p string) string {
 }
 
 func makeKickstartSudoersPost(names []string) string {
+       return ""
+
        echoLineFmt := `echo -e "%[1]s\tALL=(ALL)\tNOPASSWD: ALL" > "/etc/sudoers.d/%[1]s"
 chmod 0440 /etc/sudoers.d/%[1]s`
 
```
and no (unit) test fails afaict. 

But again, not a blocker, it's what I would love to see and it's fine if I don't see it :)